tests/php/SudoModeControllerTest.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace SilverStripe\Admin\Tests;
+
+use PHPUnit_Framework_MockObject_MockObject;
+use SilverStripe\Admin\SudoModeController;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Session;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\FunctionalTest;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\SecurityToken;
+use SilverStripe\Security\SudoMode\SudoModeService;
+use SilverStripe\Security\SudoMode\SudoModeServiceInterface;
+
+class SudoModeControllerTest extends FunctionalTest
+{
+    protected $usesDatabase = true;
+
+    /**
+     * @var bool
+     */
+    private $securityTokenEnabled;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->securityTokenEnabled = SecurityToken::is_enabled();
+        SecurityToken::disable();
+
+        $memberID = $this->logInWithPermission();
+
+        /** @var Member $member */
+        $member = Member::get()->byID($memberID);
+        $member->changePassword('0p3nS3samE!');
+    }
+
+    protected function tearDown(): void
+    {
+        if ($this->securityTokenEnabled) {
+            SecurityToken::enable();
+        }
+
+        parent::tearDown();
+    }
+
+    public function testNotLoggedIn()
+    {
+        $this->logOut();
+
+        $this->autoFollowRedirection = false;
+        $response = $this->get(SudoModeController::singleton()->Link('check'));
+        $this->autoFollowRedirection = true;
+
+        // Should redirect to login page
+        $this->assertSame(302, $response->getStatusCode());
+        $this->assertStringEndsWith(
+            Controller::normaliseTrailingSlash('/Security/login?BackURL=%2Fadmin%2Fsudomode%2Fcheck'),
+            $response->getHeader('Location'),
+            'Should not be able to check sudo mode when not logged in'
+        );
+    }
+
+    public function testLoggedInNotExpired()
```

Comment:
When we login, sudo mode is automatically enabled. We login in `setUp()`.
This used to be `assertFalse()` in `testCheckReturnsFalse()` because sudo mode used to only be activated if logging in using a form - which was inconsistent behaviour.
The behaviour is now to activate sudo mode when you log in _regardless_ of how you log in - see the framework PR.